src/models/Links.php

```diff
@@ -93,6 +94,74 @@ public function readAll(): array
     }
 
     /**
+     * Get related entities
+     *
+     * @param string $type Type of related entities (items or experiments)
+     * @return array
+     */
+    public function readRelated(string $type): array
+    {
+        if (!($type === 'items' or $type === 'experiments')) {
+            throw new ImproperActionException(_('Internal error.'));
+        }
+
+        $sql = "SELECT $type.id AS entityid, {$type}_links.id AS linkid, $type.title";
```

Comment:
strings are single quote and concatenation is used (see rest of the code)

Reply:
This is unfortunately a bit inconsistent in the rest of the code although I see that you and other contributors try to stick to single quotes.
However, I tried to simplify the SQL query by avoiding concatenation and use `sprintf()` instead as e.g. in `AbstractEntity->getReadSqlBeforeWhere()`. If you insist on concatenation I can change the three remaining occurrence of course and revert the sprintf.

https://github.com/elabftw/elabftw/blob/7f079ef93682dec7ddfd6b883f13b194673814a1/src/models/AbstractEntity.php#L835-L838

Reply:
sprintf is fine